internal/core/src/segcore/SegmentInterface.cpp

```diff
@@ -62,36 +62,41 @@ SegmentInternalInterface::FillTargetEntry(const query::Plan* plan, SearchResult&
                        "Primary key field is not INT64 type");
             bulk_subscript(key_offset, results.ids_.data(), size, blob.data());
         }
-        blobs.emplace_back(std::move(blob));
-        element_sizeofs.push_back(sizeof(int64_t));
+        results.ids_data_ = std::move(blob);
```

Comment:
unnecessary std::move.
if ids_data_ is not in use, just delete all of this block

Reply:
what do you mean "delete all of this block"?

Reply:
old filed named `ids_`, ids_data is the new ids field, so yes, it is in use

Reply:
> what do you mean "delete all of this block"?

all related code about ids_data.

Reply:
what 's the purpose of ids_data?

Reply:
> what 's the purpose of ids_data?

decouple the ids field from row_data/column_data